mood-analysis.rb

```diff
@@ -28,5 +33,45 @@ def analyze_mood(words)
   "05/11 Yay, yay, yay! I'm having a awfuly great day."
 ]
 
-puts analyze_mood(text[0])
-puts analyze_mood(text[1])
+text.each do |element|
+  date = element[0..4]
+  smiley = analyze_mood(element)
+  puts date + " "+ smiley
+end
+
+def happy_days(entries)
+  happy_entry = 0
+  general_entry = 0
+  #keep track of # of happy days and entries for each entry
+  entries.map do |entry|
+    happy_entry += 1 if analyze_mood(entry) == ":-)"
+    general_entry += 1
+    break if happy_entry == 3
```

Comment:
There's no reason to just break here instead of just putting the return right here